stdlib/public/Concurrency/Actor.cpp

```diff
@@ -1756,31 +1756,31 @@ void swift::swift_defaultActor_deallocateResilient(HeapObject *actor) {
                       metadata->getInstanceAlignMask());
 }
 
-// TODO: most likely where we'd need to create the "proxy instance" instead? (most likely remove this and use swift_distributedActor_remote_create instead)
-void swift::swift_distributedActor_remote_initialize(DefaultActor *_actor) { // FIXME: remove distributed C++ impl not needed?
-  auto actor = asImpl(_actor);
-  actor->initialize(/*remote=*/true);
-}
+OpaqueValue*
+swift::swift_distributedActor_remote_initialize(const Metadata *actorType) {
+  auto *classMetadata = actorType->getClassObject();
 
-// TODO: missing implementation of creating a proxy for the remote actor
-OpaqueValue* swift::swift_distributedActor_remote_create(OpaqueValue *identity,
-                                                         OpaqueValue *transport) {
-  assert(false && "swift_distributedActor_remote_create is not implemented yet!");
-}
+  // TODO(distributed): make this allocation smaller
+  HeapObject *alloc = swift_allocObject(classMetadata,
+                           classMetadata->getInstanceSize(),
+                           classMetadata->getInstanceAlignMask());
 
-void swift::swift_distributedActor_destroy(DefaultActor *_actor) { // FIXME: remove distributed C++ impl not needed?
-  // TODO: need to resign the address before we destroy:
-  //       something like: actor.transport.resignIdentity(actor.address)
+  // TODO(distributed): if we are going to keep the remote flag in the header,
+  //                    allocate that header and mark / register this as being remote instance.
+
+  return reinterpret_cast<OpaqueValue*>(alloc);
+}
 
-  // FIXME: if this is a proxy, we would destroy a bit differently I guess? less memory was allocated etc.
+void swift::swift_distributedActor_destroy(DefaultActor *_actor) {
+  // FIXME(distributed): if this is a proxy, we would destroy a bit differently I guess? less memory was allocated etc.
```

Comment:
Good point. That size optimization I think should be done later, once we have a stable implementation.

My guess is that `swift_allocObject` does _not_ rely on the type metadata to remember the size of the allocation, and that info is stored in a separate header like other `malloc` implementations, but we'll need to double-check when implementing the optimization.

Reply:
Agreed, I've added a radar for it too